src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -168,6 +168,11 @@ def validate_client_parameters(cmd, namespace):
     if n.sas_token:
         n.sas_token = n.sas_token.lstrip('?')
 
+    # account name with secondary
+    if n.account_name.endswith('-secondary'):
+        n.location_mode = 'secondary'
+        n.account_name = n.account_name[:-10]
+
```

Comment:
Consider this as a temporary solution. Exposing `--location-mode` to users would be more clear.

Reply:
because in track 2, the accepted params for client is account url not account name and location mode. Current design could fix with track2 SDk, too.